install_script_op_worker2.sh

```diff
@@ -164,6 +164,64 @@ if [ -n "$DD_INSTALL_ONLY" ]; then
     no_start=true
 fi
 
+# sources
+source_splunk_hec_address=
+if [ -n "$DD_OP_SOURCE_SPLUNK_HEC_ADDRESS" ]; then
+    source_splunk_hec_address=$DD_OP_SOURCE_SPLUNK_HEC_ADDRESS
+fi
```

Comment:
Maybe I'm missing something, as I haven't been watching these scripts. What does this construction achieve that a simple assignment doesn't?

Also note, since the RHS is unquoted, this script will misbehave if it contains spaces or other special characters.

Reply:
As far as I'm aware, a lot of "cargo culting" from the existing implementations. Even if it's unneeded I'd opt to keep it in-line with the other usages, and we can revisit with a larger fix/refactor.

Reply:
I'm not the OG author but after reviewing the script, I suspect this was intentionally done for things like the DD_API_KEY and pipeline_id , so that it would be easier to pass around the shorter local script var (apikey, op_pipeline_id) , than check the actual env var each place its used.

Because the component env vars that are being added here are not used in as many places, I believe we could remove this extraction to local vars and just use direct down circa line 538.

HOWEVER- there is talk in the slack thread about trying to get this done today... so if this works, in the sake of time I think we should proceed since we are planning to open a ticket to follow up with a more robust solution anyway.

WDYT?

I can add quotes to the RHS